lib/galaxy/tool_util/xsd/galaxy.xsd

```diff
@@ -2739,6 +2751,123 @@ $attribute_list::5
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="channels" type="xs:integer" use="optional">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">The required number of channels of the image (e.g., 1 for grayscale, 3 for RGB, 4 for RGBA images).</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+  </xs:complexType>
+  <xs:complexType name="AssertImageHasIntensities">
+    <xs:annotation>
+      <xs:documentation xml:lang="en"><![CDATA[
+Asserts the image intensities comply with the specified constraints.
+
+```xml
+<image_has_intensities mean_intensity="0.83" center_of_mass="511.07, 223.34" eps="0.01" />
+```
+
+$attribute_list::5
+]]>
+      </xs:documentation>
+    </xs:annotation>
+    <xs:attribute name="mean_intensity" type="xs:float" use="optional">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">The required mean value of the image intensities.</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="mean_intensity_min" type="xs:float" use="optional">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">A lower bound of the required mean value of the image intensities.</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="mean_intensity_max" type="xs:float" use="optional">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">An upper bound of the required mean value of the image intensities.</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="center_of_mass" type="xs:string" use="optional">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">The required center of mass of the image intensities (horizontal and vertical coordinate, separated by a comma).</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="eps" type="xs:float" use="optional" default="0.01">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">The absolute tolerance to be used for the ``mean_intensity`` and ``center_of_mass`` assertions (defaults to ``0.01``).</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="channel" type="xs:integer" use="optional">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">Restricts the ``mean_intensity``, ``mean_intensity_min``, ``mean_intensity_max``, and ``center_of_mass`` assertions to a specific channel of the image (where the value ``0`` corresponds to the first image channel).</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+  </xs:complexType>
+  <xs:complexType name="AssertImageHasLabels">
+    <xs:annotation>
+      <xs:documentation xml:lang="en"><![CDATA[
+Asserts the labels in a label image comply with the specified constraints.
+
+```xml
+<number_of_objects number_of_objects="187" mean_object_size="111.87" eps="0.01" exclude_labels="0" />
```

Comment:
I think assertions like `has_image_width` or `has_image_center_of_mass` are rather unintuitive[^1], but I agree that his scheme is more consistent with the other assertions currently present in Galaxy.

So I guess I will change it.

[^1]: In spoken language, you want to assert that *"the image has a width"* and *"the image has a center of mass"*, not that the output *"has an image center"* of something.

Reply:
Done this in 00485ccc5294815c06e1d4d51d92d487f4133c65.

Reply:
Sorry for the delay and nitpicking: What I meant here is that it would be even more consistent with the present assertions to have 

```
<has_image_width witdth="" ...
```

instead of 

```
<has_image_witdth value=""
```

We [recently](https://github.com/galaxyproject/galaxy/pull/17490) also implemented this for `has_size`. 

Apart from this I'm super happy with the PR.

Reply:
Done this in 5964f72dfcb3092ecb3eac1037daa6dae099b2c0 and 8a81593c5372d1e2b6031e6ae62a820f4ef6857c.

Thanks for your suggestions!